subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/JavaToolchainInput.java

```diff
@@ -32,6 +33,7 @@ public JavaToolchainInput(JavaToolchainSpec spec) {
         this.implementation = spec.getImplementation().get().toString();
     }
 
+    @Optional
```

Comment:
Shall we not add this as an input here and always use the one from the toolchain itself which is now an input?